Bio/SearchIO/BlastIO/blast_xml.py

```diff
@@ -329,11 +329,11 @@ def _parse_qresult(self):
                     if hit:
                         # need to keep track of hit IDs, since there could be duplicates,
                         if hit.id in key_list:
-                            warnings.warn("Adding hit with BLAST-generated ID "
-                                    "%r since hit ID %r is already present "
-                                    "in query %r. Your BLAST database may contain "
+                            warnings.warn("Renaming hit ID %r to a BLAST-generated ID "
```

Comment:
There is only one BLAST-generated hit for each hit indeed. However this scenario occurs when the same hit ID exists more than once, and for each of those duplicate hit IDs there can be more than one ID that BLAST generates (depending on how many times they occur).